pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -235,62 +226,71 @@ func (m *KubeIPResolverInstance) enrich(ev any) {
 	additionalInfo, _ := ev.(KubeNetworkInformation)
 	containerInfo, _ := ev.(operators.ContainerInfoGetters)
 
-	additionalInfo.SetRemoteKind(types.RemoteKindOther)
-
 	pods := m.manager.k8sInventory.GetPods()
 	foundLocal := false
-	foundRemote := false
-	remoteIP := additionalInfo.GetRemoteIP()
+	foundRemote := 0
+	remoteIPs := additionalInfo.GetRemoteIPs()
+	endpoints := make([]types.EndpointDetails, len(remoteIPs))
+	for j := range endpoints {
+		// initialize to this default value if we don't find a match
+		endpoints[j].Kind = types.RemoteKindOther
+	}
 
 	for i, pod := range pods.Items {
-		if foundLocal && foundRemote {
+		if foundLocal && foundRemote == len(remoteIPs) {
 			return
 		}
 
 		if pod.Namespace == containerInfo.GetNamespace() && pod.Name == containerInfo.GetPod() {
 			foundLocal = true
-			additionalInfo.SetPodIP(pod.Status.PodIP)
-			additionalInfo.SetPodHostIP(pod.Status.HostIP)
-			additionalInfo.SetPodLabels(pod.Labels)
-
+			owner := ""
 			// When the pod belongs to Deployment, ReplicaSet or DaemonSet, find the
 			// shorter name without the random suffix. That will be used to
 			// generate the network policy name.
 			if pods.Items[i].OwnerReferences != nil {
 				nameItems := strings.Split(pods.Items[i].Name, "-")
 				if len(nameItems) > 2 {
-					additionalInfo.SetPodOwner(strings.Join(nameItems[:len(nameItems)-2], "-"))
+					owner = strings.Join(nameItems[:len(nameItems)-2], "-")
 				}
 			}
+			additionalInfo.SetLocalPodDetails(owner, pod.Status.HostIP, pod.Status.PodIP, pod.Labels)
 		}
 
 		if pod.Spec.HostNetwork {
 			continue
 		}
-		if pod.Status.PodIP == remoteIP {
-			foundRemote = true
-			additionalInfo.SetRemoteKind(types.RemoteKindPod)
-			additionalInfo.SetRemoteName(pod.Name)
-			additionalInfo.SetRemoteNamespace(pod.Namespace)
-			additionalInfo.SetRemotePodLabels(pod.Labels)
+		for j, remoteIP := range remoteIPs {
+			if pod.Status.PodIP == remoteIP {
+				foundRemote++
+				endpoints[j].Kind = types.RemoteKindPod
+				endpoints[j].Name = pod.Name
+				endpoints[j].Namespace = pod.Namespace
+				endpoints[j].PodLabels = pod.Labels
+			}
 		}
 	}
-
-	if foundRemote {
+	if foundRemote == len(remoteIPs) {
+		additionalInfo.SetEndpointsDetails(endpoints)
 		return
 	}
 
 	svcs := m.manager.k8sInventory.GetSvcs()
 
 	for _, svc := range svcs.Items {
-		if svc.Spec.ClusterIP == remoteIP {
-			additionalInfo.SetRemoteKind(types.RemoteKindService)
-			additionalInfo.SetRemoteName(svc.Name)
-			additionalInfo.SetRemoteNamespace(svc.Namespace)
-			additionalInfo.SetRemotePodLabels(svc.Labels)
-			break
+		for j, remoteIP := range remoteIPs {
+			if svc.Spec.ClusterIP == remoteIP {
+				endpoints[j].Kind = types.RemoteKindService
+				endpoints[j].Name = svc.Name
+				endpoints[j].Namespace = svc.Namespace
+				endpoints[j].PodLabels = svc.Labels
+			}
 		}
 	}
+
+	if foundRemote > 0 {
```

Comment:
If we can't resolve any IPs to K8s resources, shouldn't `.SetEndpointDetails(endpoints)` still be called?
So the tracer gets the info that all IPs are of type `types.RemoteKindOther`